### PR TITLE
Add: distinguish vaults visible as admin vs visible as members

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

Distinguish the vaults accessible as members and the one you can see because you are an admin.

https://github.com/user-attachments/assets/4179cd6a-2bd5-4466-a764-1cc71593eb0c

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/pages/api/w/[wId]/vaults/index.ts [hunk 1]
```diff
@@ -31,7 +31,25 @@ async function handler(
 
   switch (req.method) {
     case "GET":
-      const vaults = await VaultResource.listWorkspaceVaults(auth);
+      const { role } = req.query;
```
Comment (spolu) [inline front/pages/api/w/[wId]/vaults/index.ts:34]: Should we check first that the user is admin if role is admin to return a proper authorization error?

File: front/lib/resources/vault_resource.ts [hunk 2]
```diff
@@ -156,14 +156,18 @@ export class VaultResource extends BaseResource<VaultModel> {
...
+
+  static async listWorkspaceVaultsAsAdmin(
+    auth: Authenticator
   ): Promise<VaultResource[]> {
     const vaults = await this.baseFetch(auth);
     return vaults.filter(
-      (vault) =>
-        (auth.isAdmin() && adminsBypassACL) ||
-        auth.hasPermission([vault.acl()], "read")
+      (vault) => auth.isAdmin() || auth.hasPermission([vault.acl()], "read")
```
Comment (flvndvd) [inline front/lib/resources/vault_resource.ts:170]: Do we need the `auth.hasPermission([vault.acl()], "read")`? I guess `auth.isAdmin()` should be enough, right?

**Review by flvndvd — APPROVED** [review]
LGTM

File: front/pages/api/w/[wId]/vaults/index.ts (hunk 1 above)
Comment (Fraggle) [reply front/pages/api/w/[wId]/vaults/index.ts:34]: I didn't want to do that initially to avoid conditionnal code in the client but I will change.

**Pull request merged**